trpcultivate_phenotypes/tests/src/Functional/ConfigOntologyTermsTest.php

```diff
@@ -0,0 +1,161 @@
+    // to creation of Ontology and Terms will show a warning message.
+    $this->drupalGet('admin/tripal/extension/tripal-cultivate/phenotypes/ontology');
+    $session = $this->assertSession();
+    $session->statusCodeEquals(200);
+    $session->pageTextContains('Warning message');
+
+    // Tripal Jobs to create/insert terms and setup genus ontology configuration
+    // are created on install of tripalcultivate_phenotypes. The job may execute or not
+    // but this block will create them manually.
+    $test_insert_genus = [
+      'Lens',
+      'Cicer'
+    ];
+
+    $ins_genus = "
+      INSERT INTO {1:organism} (genus, species, type_id)
+      VALUES
+        ('$test_insert_genus[0]', 'culinaris', 1),
+        ('$test_insert_genus[1]', 'arientinum', 1)
+    ";
+
+    $this->chado->query($ins_genus);
+
+    // Load genus ontology.
+    $service_genusontology = \Drupal::service('trpcultivate_phenotypes.genus_ontology');
+    $service_genusontology->loadGenusOntology();
+
+    // Install all default terms.
+    $service_terms = \Drupal::service('trpcultivate_phenotypes.terms');
+    $service_terms->loadTerms();
+
+    // Access Ontology and Terms configuration page.
+    $this->drupalGet('admin/tripal/extension/tripal-cultivate/phenotypes/ontology');
+    $session = $this->assertSession();
+
+    $session->statusCodeEquals(200);
+    $session->pageTextContains('Configure Tripal Cultivate Phenotypes: Ontology Terms');
+
+    $genus_ontology = $service_genusontology->defineGenusOntology();
+    
+    $null_value = 1;
+    $values_genusontology = [];
+    
+    foreach($genus_ontology as $genus => $vars) {
+      foreach($vars as $i => $config) {
+        $fld_name = $genus . '_' . $config;
+        // Test if each genus has a trait, unit, method, db and crop ontology field.
+        $session->fieldExists($fld_name);
+
+        // set to Null (id: 1) all genus ontology configuration.
+        $values_genusontology[ $fld_name ] = $null_value;
+      }
+    }
+
+    // Update default values.
+    $this->submitForm($values_genusontology, 'Save configuration');
+    $session->pageTextContains('The configuration options have been saved.');
+    
+    foreach(array_keys($values_genusontology) as $fld) {
+      // If all config got null set as the value.
+      $session->fieldValueEquals($fld, $null_value);
+    }
+    
+    // Allow new trait.
+    $allow_new = 'allow_new';
+    $session->fieldExists($allow_new);
+    // Update allow new.
+    $this->submitForm([$allow_new => FALSE], 'Save configuration');
+    $session->pageTextContains('The configuration options have been saved.');
+    $session->fieldValueEquals($allow_new, FALSE);
+
+    // Terms.
+    $null_value = 'null (null:local:null)';
+    $terms = $service_terms->defineTerms();
+    $values_terms = [];
+
+    foreach($terms as $config => $prop) {
+      // Test each term has an autocomplete field.
+      $session->fieldExists($config);
+
+      $values_terms[ $config ] = $null_value;
```

Comment:
Same goes for this one (i.e. do not set them all to the same thing) although it's less important in a list then it is in a grid.

Reply:
same above, used generated terms.